sphinxgallery/gen_rst.py

```diff
@@ -319,14 +322,17 @@ def make_thumbnail(in_fname, out_fname, width, height):
         import Image
     img = Image.open(in_fname)
     width_in, height_in = img.size
-    scale_w = width / float(width_in)
-    scale_h = height / float(height_in)
+    scale_w = max_width / float(width_in)
+    scale_h = max_height / float(height_in)
 
-    if height_in * scale_w <= height:
+    if height_in * scale_w <= max_height:
         scale = scale_w
     else:
         scale = scale_h
 
+    if scale >= 1.0 and in_fname == out_fname:
+        return
```

Comment:
The change of scale_image is more an import of the mayavi branch pull request. I do find it more convenient to have it like this than the old make_thumbnail function and it is something I started working early on for this new release start.
Why this last line? It is actually the work of @agramfort